src/mds/CDir.cc

```diff
@@ -2695,6 +2695,17 @@ bool CDir::contains(CDir *x)
   }
 }
 
+bool CDir::can_rep() const
+{
+  if (!is_rep()) 
+    return true;
+
+  unsigned mds_num = mdcache->mds->get_mds_map()->get_num_mds(MDSMap::STATE_ACTIVE);
+  if ((mds_num - 1) != get_replicas().size()) 
```

Comment:
What about the edge case where the user decides to change max_mds intentionally. Then, won't this condition be true every-time that directory is hit and lead to repeated broadcasts for replication? @ukernel and @batrick Please do look into this too.

Reply:
This `can_rep()` is used to decide whether this dir needs to be replicated. If true, then `send_dir_updates` will figure out which MDSes haven't contained this dir's replica. So this dir will be broadcasted to those MDSes only.

Reply:
Yes that's right. But my concern here is that if the number of MDS is changed manually by the user (ceph fs set <fs_name> max_mds n-1), then can_rep() will always be true after that, unless the correct replica state is updated in the auth mds when max mds is changed (which I don't think is the case).

Reply:
Do you mean this will cost some overhead if can_rep() is always true in the case of reducing max mds, although no actual replication would happen in send_dir_updates()?

If so, how about modifying the checks to:

`if ((mds_num - 1) > get_replicas().size()) `

Reply:
Yes I think this would be better. Plus I think it does send a broadcast dir update message to every active mds .See : https://github.com/ceph/ceph/blob/a9fdaa1523939e9e329aefdc8d131a869974bff8/src/mds/MDBalancer.cc#L1239
So it does cost a significant overhead.

Reply:
@sidharthanup Done. Pls review again.